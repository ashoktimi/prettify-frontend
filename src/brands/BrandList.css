/* Bandslist style for navbar  */
.BrandList-column {
  display: flex;
  justify-content:space-around;
  position: absolute;
  flex-wrap: wrap;
  left: 0;
  right: 0;   
  margin: auto;
  list-style: none;
  background-color: rgb(195, 206, 182);
  padding:10%;
  z-index: 2;
}

.BrandList-li{
  margin-bottom: 1.5rem;  
  margin-right: 2rem;
}
.BrandList-link {
  color: black;
  text-decoration: none;
  text-transform: capitalize;
  font-family: 'Heebo', sans-serif;
  font-size: 25px;
}
  
.BrandList-link:hover {
  text-decoration: underline;
}

/* Bandslist style for toggler  */

.Brandlist-toggler-ul {
  list-style: none;
}

.Brandlist-toggler-ul li {
  margin: 2.3rem 0;
  border-bottom: .5px solid rgb(233, 227, 227);
}
.Brandlist-toggler-ul a {
  text-transform: capitalize;
  font-weight: bold;
  text-decoration: none; 
  color: #333;
}
.Brandlist-toggler-ul a:hover{
  color: purple;
}

  
