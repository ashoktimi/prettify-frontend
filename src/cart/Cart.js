import { useEffect, useContext } from "react";
import { useDispatch, useSelector } from "react-redux";
import UserContext from "../auth/userContext";
import EmptyCart from "./EmptyCart";

import { RiDeleteBinLine } from 'react-icons/ri';
import {
  addToCart,
  clearCart,
  decreaseCart,
  getTotals,
  removeFromCart,
} from "./cartSlice";
import { Offcanvas, Stack, Button } from "react-bootstrap";

const Cart = () => {
  const cart = useSelector((state) => state.cart);
  const dispatch = useDispatch();
  const { showOffcanvas, handleToggleOffcanvas } = useContext(UserContext);


  useEffect(() => {
    dispatch(getTotals());
  }, [cart, dispatch]);

  const handleAddToCart = (product) => {
    dispatch(addToCart(product));
  };
  const handleDecreaseCart = (product) => {
    dispatch(decreaseCart(product));
  };
  const handleRemoveFromCart = (product) => {
    dispatch(removeFromCart(product));
  };
  const handleClearCart = () => {
    dispatch(clearCart());
  };

  console.log((cart.cartItems).length)

  return (
    <>
    <Offcanvas show={showOffcanvas} onHide={handleToggleOffcanvas} placement="end">
      <Offcanvas.Header closeButton>
        <Offcanvas.Title style={{left:0, right:0, margin:'auto'}}>Shopping Cart</Offcanvas.Title>
      </Offcanvas.Header>
      <Offcanvas.Body>
    {(cart.cartItems).length?
    <>
    <Stack gap={3}>
      {cart.cartItems.map(item => (
        <Stack gap={2}>
          <Stack direction="horizontal">
          <img
            src={item.image_link}    
            style={{ width: "125px", height: "115px", objectFit: "cover" }}
            alt={''}
          />
            {item.name}{" "}${(item.price)}
            {item.cartQuantity > 1 && (
                <span className="text-muted" style={{ fontSize: ".85rem", letterSpacing:'30px'}}>
                  x{item.cartQuantity }
                </span>
            )}
            {item.cartQuantity > 1?
            <div style={{display: "flex", justifyContent: 'flex-end'}}> 
                ${(item.price * item.cartQuantity)}
            </div> : null
            }
          </Stack>        
          
            <Stack direction="vertical" >         

              <div style={{display: "flex", justifyContent: 'center'}}>
                <button onClick={() => handleDecreaseCart(item)} style={{border:'none', backgroundColor:'white', marginRight: '15px'}}>
                  -
                </button>
                <div style={{ marginRight: '15px'}}>{item.cartQuantity}</div>
                <button onClick={() => handleAddToCart(item)} style={{border:'none', backgroundColor:'white',  marginRight: '15px'}}> + </button>
               
              <Button
                variant="light"
                onClick={() => handleRemoveFromCart(item)}
                style={{backgroundColor: 'white'}}               
              >
                <RiDeleteBinLine />
            </Button>
            </div>   
            </Stack>
        </Stack>

      ))}
      <Button 
        className="ms-auto"
        variant="outline-danger"
        size="sm" 
        onClick={()=> handleClearCart()} 
        style={{width:'fit-content'}}  
       
      >
        Clear Cart
      </Button>
    </Stack>


      <div className="subtotal"style={{ display: 'flex', justifyContent: 'space-between', marginTop: '1rem'}}>
        <span style={{marginLeft: '15px'}}>Subtotal : </span>
        <span className="amount">${cart.cartTotalAmount}</span>     
      </div>
      <Button href="/checkout" variant="warning" style={{ margin: '5rem'}}>Check Out</Button>

      </>
      :
      <EmptyCart />
      }
      </Offcanvas.Body>
    </Offcanvas>
    </>
  )

};

export default Cart;