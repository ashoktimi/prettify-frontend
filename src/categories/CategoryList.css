/* css property for categorylist in navbar */
.CategoryList-column {
  display: flex;
  justify-content: center;
  position: absolute;
  flex-wrap: wrap;
  left: 0;
  right: 0;   
  margin: auto;
  list-style: none;
  background-color: rgb(195, 206, 182);
  padding:10%;
  z-index: 2;
}
.CategoryList-li{
  margin-bottom: 1.5rem;  
  margin-right: 3rem;
}
  .CategoryList-link {
    color: black;
    text-decoration: none;
    text-transform: capitalize;
    font-family: 'Heebo', sans-serif;
    font-size: 25px;
}
    
.BrandList-link:hover {
  text-decoration: underline;
}

/* css property for categorylist in offcanvas toggler */

.Categorylist-toggler-ul {
  list-style: none;
}

.Categorylist-toggler-ul li {
  margin: 3.7rem 0;
  border-bottom: .5px solid rgb(233, 227, 227);
}
.Categorylist-toggler-ul a {
  text-transform: capitalize;
  font-weight: bold;
  text-decoration: none; 
  color: #333;
}
.Categorylist-toggler-ul a:hover{
  color: purple;
}