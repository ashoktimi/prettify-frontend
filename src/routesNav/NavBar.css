
.NavBar-link {
    color: #333333;
    text-decoration:none;
    margin-left: 1.5rem;
    font-weight: bold;
    font-family: 'Heebo', sans-serif;
    margin-top: 15px;
    font-size: 20px;
} 

.NavBar-dropdown{
    font-weight: bold;
    font-family: 'Heebo', sans-serif;
    margin-left: 1.5rem;
}

.NavBar-dropdown > .dropdown-toggle::after {
  display: none;
}

.nav-link{
    color: #333333;
    padding-top: 5px;
    margin-top: 0;
    font-size: 20px;
}

.nav-item{
  margin-top: 10px;
}
.NavBar-dropdown-link{
    color: #333333;
    font-family: 'Heebo', sans-serif;
    text-decoration: none;
    text-transform: capitalize;
}

.Navbar-bag{
    color:black;
    text-decoration: none;
    margin-right: 15px;
    margin-left: 2.5rem;
    margin-top: 10px;
}

.Navbar-bag:hover{
  color: black;
}

.bag-quantity {
  border-radius: 50%;
  font-size: 20px;
  font-weight: 700;
}
.Profile-dropdown{
  padding-top: 5px;
}
.Profile-dropdown > .dropdown-toggle::after{
  display: none;
}

.Navbar-logout{
  color: #333333;
  text-decoration: none;
  font-weight: bold;
  font-size: 20px;
  font-family: 'Heebo', sans-serif;
  margin-top: 15px;
  margin-left: 15px;
}

.Navbar-logout:hover{
  color: black;
}
