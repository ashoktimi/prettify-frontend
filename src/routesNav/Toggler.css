.NavBar-name-div{
    margin: auto;
    left: 0;
    right: 0;
}
.NavBar-name{
    font-family: 'Caveat', cursive;
    color: purple;
    letter-spacing: 0.5rem;
    font-size: 3rem;
    text-decoration: none;
}
.NavBar-name:hover{
  color: purple;
}
.navbar-toggler{
  color:rgb(243, 224, 230);
  border: none;
}

.NavBar-auth_Btn{
  margin-left: 1rem;
  color: #333333;
  text-decoration: none;
  font-weight: bold;
  font-family: 'Heebo', sans-serif;
} 
.Toggler-logout{
  margin-right: 10rem;
  color: #333333;
  text-decoration: none;
  font-weight: bold;
  font-family: 'Heebo', sans-serif;
  margin-left: 2.5rem;
  margin-top: 10px;
  font-size: 20px;
} 
.Toggler-logout:hover{
  color: black;
}

.Toggler-bag{
  color:black;
  text-decoration: none;
  margin-left: 2.5rem;
  margin-bottom: 1.5rem;
}
.Toggler-bag:hover{
  color: black;
}



/* for toggler elements */
.Toggler-h3{
  left: 0;
  right: 0;
  margin: auto;
  color: black;
  font-size: 2rem;
}

.Toggler-ul {
  list-style: none;
}

.Toggler-ul li {
  margin: 3rem 0;
  border-bottom: .5px solid rgb(233, 227, 227);
}

.Toggler-ul a {
  font-weight: bold;
  text-decoration: none;  
  color: #333;
}
.Toggler-ul a:hover {
  color: #9c6a8d;
}
