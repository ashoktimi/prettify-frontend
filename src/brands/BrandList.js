import React from "react";
import { Link } from "react-router-dom";
import './BrandList.css'

/** Show page with list of brands.
 *
 * On mount, loads brands from API. 
 *
 * This is routed to at /brands
 */
const BrandList = ({ brands, handleHover }) =>{
    return (
      <>
      {brands && handleHover ?
        <div >
          <ul className="BrandList-column">
            {brands.map(b => (            
              <li className="BrandList-li" key={b.id} onClick={handleHover}>
                 <Link className="BrandList-link" to={`/brands/${(b.name).trim().replace(/ /g, "-")}/${b.id}`}>
                   {b.name}
                 </Link>  
              </li>       
            ))}
          </ul> 
        </div>
        :
        <ul className="Brandlist-toggler-ul">
          {brands.map(brand => (
          <li key={brand.id}><Link to={`/brands/${brand.name}/${brand.id}`}>{brand.name}</Link></li>
          ))}
        </ul>  
      }
      </>
    );
}

export default  BrandList;


