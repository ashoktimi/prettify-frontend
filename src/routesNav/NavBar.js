import React, { useState, useEffect, useContext } from 'react';
import { Link } from 'react-router-dom';
import { useSelector } from "react-redux";
import PrettifyApi from "../api/api";
import BrandList from "../brands/BrandList";
import CategoryList from "../categories/CategoyList";
import TagList from "../tags/TagList";
import Toggler from './Toggler';
import Cart from "../cart/Cart";
import UserContext from "../auth/userContext";
import { CgProfile } from 'react-icons/cg'
import { AiOutlineShoppingCart } from 'react-icons/ai';
import 'bootstrap/dist/css/bootstrap.css';
import { Container,Navbar, Nav, NavDropdown } from 'react-bootstrap';
import './NavBar.css';


function NavBar({ logout }) {
  const { currentUser, handleToggleOffcanvas} = useContext(UserContext);
  const { cartTotalQuantity } = useSelector((state) => state.cart);
  // Merge similar state variables into a single object
  const [navBarData, setNavBarData] = useState({
    brands: [],
    categories: [],
    tags: [],
    types: { eyes: [], lips: [], face: [] },
    isHovered: false,
    isCategoryHovered: false,
    isTagHovered: false,
    isProfileHovered: false,
  });

  useEffect(() => {
    async function fetchData() {
      const { brands } = await PrettifyApi.getBrands();
      const { categories } = await PrettifyApi.getCategories();
      const { tagLists: tags } = await PrettifyApi.getTaglists();

      setNavBarData((prevData) => ({
        ...prevData,
        brands,
        categories,
        tags,
      }));
    }

    fetchData();
  }, []);

  useEffect(() => {
    async function fetchTypes(type) {
      const { types } = await PrettifyApi.getTypes(type);
      setNavBarData((prevData) => ({
        ...prevData,
        types: { ...prevData.types, [type]: types },
      }));
    }

    fetchTypes('eyes');
    fetchTypes('lips');
    fetchTypes('face');
  }, []);

  // Extracted hover handlers to remove duplication
  const handleHover = (stateVar) => {
    setNavBarData((prevData) => ({
      ...prevData,
      [stateVar]: !prevData[stateVar],
    }));
  };

  function renderNavItem(type) {
    return (
      <NavDropdown className="NavBar-dropdown" title={type.toUpperCase()} >
        {navBarData.types[type].map((t) => (
          <NavDropdown.Item key={t.name}>
            <Link to={`/types/${type}/${t.name}`} className="NavBar-dropdown-link">
              {(t.name).trim().replace("_", " ")}
            </Link>
          </NavDropdown.Item>
        ))}
      </NavDropdown>
    );
  }


  return (
    <>
      <Navbar sticky='top' style={{ backgroundColor: '#C16FB8', padding: '1rem', height: '4rem'}}/>
      <Toggler logout={logout} {...navBarData} renderNavItem={renderNavItem} />
      <Cart />
      {currentUser && (
       <Navbar sticky='top' expand="lg">
        <Container >
        {/* <Navbar.Toggle aria-controls="responsive-navbar-nav" disabled/> */}
        <Navbar.Collapse id="responsive-navbar-nav" >
 
          <Nav className="mr-auto" style={{ left:0, right:0, margin:'auto'}}>
            <Link to="/products" className="NavBar-link">ALL</Link>
            <Link to="/types/nail/nail_polish" className="NavBar-link">NAIL</Link>
            {renderNavItem('eyes')}
            {renderNavItem('lips')}
            {renderNavItem('face')}
            <Nav.Item onMouseEnter={() => handleHover('isHovered')} onMouseLeave={() => handleHover('isHovered')} style={{paddingTop:'5px'}}>
              <Link className="NavBar-link">BRANDS</Link>
              {navBarData.isHovered && <BrandList handleHover={() => handleHover('isHovered')} brands={navBarData.brands} />}
            </Nav.Item>
            <Nav.Item onMouseEnter={() => handleHover('isCategoryHovered')} onMouseLeave={() => handleHover('isCategoryHovered')} style={{paddingTop:'5px'}}>
              <Link className="NavBar-link">CATEGORIES</Link>
              {navBarData.isCategoryHovered && <CategoryList handleCatHover={() => handleHover('isCategoryHovered')} categories={navBarData.categories} />}
            </Nav.Item>
            <Nav.Item onMouseEnter={() => handleHover('isTagHovered')} onMouseLeave={() => handleHover('isTagHovered')} style={{paddingTop:'5px'}}>
              <Link className="NavBar-link">TAGS</Link>
              {navBarData.isTagHovered && <TagList handleTagHover={() => handleHover('isTagHovered')} tags={navBarData.tags} />}
            </Nav.Item>
            <Link className="Navbar-bag" onClick={handleToggleOffcanvas}>
              <AiOutlineShoppingCart size={30}/>
              <span className="bag-quantity">{cartTotalQuantity}</span>
            </Link>
            <NavDropdown className="Profile-dropdown" title={<CgProfile />} style={{marginTop: '10px'}}>
              <NavDropdown.Item>
                <Link to={'/'}className="NavBar-dropdown-link"> My Account </Link>
              </NavDropdown.Item>
              <NavDropdown.Item>
                <Link to={'/'}className="NavBar-dropdown-link"> Order History </Link>
              </NavDropdown.Item>
              <NavDropdown.Item>
                <Link to={'/'}className="NavBar-dropdown-link"> My WishList </Link>
              </NavDropdown.Item>
            </NavDropdown>
            <Link to="/" onClick={logout} className="Navbar-logout">Logout</Link>
        </Nav>

        </Navbar.Collapse>
        </Container>
       </Navbar>
      )}
    </>
  );
}

export default NavBar;