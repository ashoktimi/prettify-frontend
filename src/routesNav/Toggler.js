import React, {useState, useContext, useEffect} from "react";
import { Link } from 'react-router-dom';
import PrettifyApi from "../api/api";
import BrandList from "../brands/BrandList";
import CategoryList from "../categories/CategoyList";
import TagList from "../tags/TagLIst";
import TypeList from "../types/TypeList";
import UserContext  from "../auth/userContext";
import 'bootstrap/dist/css/bootstrap.css';
import { AiOutlineShoppingCart } from 'react-icons/ai';
import { Container, Navbar, Offcanvas } from 'react-bootstrap'
import './Toggler.css';




const Toggler = ({ logout, brands, categories, tags }) =>{

  const { currentUser, handleToggleOffcanvas} = useContext(UserContext);
  const [showBrands, setShowBrands] = useState(false);
  const [showCategories, setShowCategories] = useState(false);
  const [showTags, setShowTags] = useState(false);
  const [type, setType] = useState([]);
  const [showTypes, setShowTypes] = useState(false);
  const [value, setValue] = useState('nail'); 
  const [typeDetail, setTypeDetail] = useState([]);


  
  useEffect(() =>{
    async function getBrands() {
    let types = await PrettifyApi.getAllTypes();     
    setType(types.types);
    }
    getBrands()   
  },[])

  const uniqueTypes = [...new Set(type.map(t => t.type))];

  useEffect(() => {
    async function fetchTypeDetail() {
      if (value) {
        const response = await PrettifyApi.getTypes(value);
        setTypeDetail(response.types);
      }
    }
    fetchTypeDetail();
  }, [value]);
  
  function handleTypeClick(event) {
    const key = event.target.getAttribute('data-key');
    setValue(key);
    setShowTypes(true);
  }

  const handleBrandsClick = () => {
    setShowBrands(true);
  };
  const handleCateoryClick = () => {
    setShowCategories(true);
  };
  const handleTagClick = () => {
    setShowTags(true);
  };
  const handleBackClick = () => {
    setShowBrands(false);
    setShowCategories(false);
    setShowTags(false);
    setShowTypes(false);
  };

    return(
        <>  
        <Navbar expand='bg' style={{padding:0}}>
          <Container fluid>
            <Navbar.Toggle aria-controls={`offcanvasNavbar-expand`}/>
            <div className="NavBar-name-div">
              {currentUser ? 
                <Link to="/products" className="NavBar-name">Prettify</Link> 
                :               
                <Link to="/" className="NavBar-name" style={{marginLeft: "10rem"}}>Prettify</Link>
              }
            </div>
            {currentUser ?
            <>
            </>
             :
            <div style={{marginRight: '5rem'}}>
              <Link className="NavBar-auth_Btn" to="/login">Login</Link>  
              <Link className="NavBar-auth_Btn" to="/signup">Signup</Link>
             </div>
             }
            <Navbar.Offcanvas
              id={`offcanvasNavbar-expand`}
              aria-labelledby={`offcanvasNavbarLabel-expand`}
              placement="start"
            >
            {currentUser ?
              <>
              <Offcanvas.Header closeButton>
                {(brands.length > 0 || categories.length || tags.length> 0 || typeDetail.length > 0) && (showBrands || showCategories || showTags || showTypes)&& (
                  <button  onClick={handleBackClick} style={{border:'none', fontSize:'25px', backgroundColor:'white'}}> {`<`} </button>
                )}
                <Offcanvas.Title id={`offcanvasNavbarLabel-expand`} style={{color:'purple', left:0, right:0, margin:'auto'}}>
                  Prettify
                </Offcanvas.Title>
              </Offcanvas.Header>
              <Offcanvas.Body>
              
              {brands.length > 0  && showBrands ? (    
               <BrandList brands={brands} />
              ) : (categories.length > 0 && showCategories) ? (
               <CategoryList categories={categories} />
              ): (tags.length > 0 && showTags) ?(
               <TagList tags={tags}/>
              ):(typeDetail.length > 0 && showTypes)?(
               <TypeList types={typeDetail}/>
              ):

              <ul className="Toggler-ul" >
                <li><Link to="/products"  >ALL</Link></li>
                {uniqueTypes.map(type => (
                  <li key={type}>
                    <Link data-key={type} onClick={handleTypeClick}>{type.toUpperCase()}</Link>
                  </li>
                ))}
                <li><Link onClick={handleBrandsClick}> BRANDS </Link></li>
                <li><Link onClick={handleCateoryClick} >CATEGORY</Link></li>
                <li><Link onClick={handleTagClick} >TAGS</Link></li>
              </ul>             
              }
              </Offcanvas.Body>
              <Link className="Toggler-bag" onClick={handleToggleOffcanvas}>
                <AiOutlineShoppingCart size={30}/>
              </Link>
              <Link  to="/" className="Toggler-logout" onClick={logout} style={{marginBottom:'35px'}} >Log out</Link>      
              </>
              : <h3 className="Toggler-h3">Please login to see.</h3>}
            </Navbar.Offcanvas>
          </Container>
        </Navbar>
      </>
    )
}

export default Toggler;