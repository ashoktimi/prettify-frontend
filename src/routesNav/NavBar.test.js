import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { UserProvider,  commonBeforeEach, commonAfterEach} from "../testUtils";
import { configureStore } from '@reduxjs/toolkit';
import { BrowserRouter } from "react-router-dom";
import cartReducer from '../cart/cartSlice';
import NavBar from "./NavBar";
import Toggler from "./Toggler";
import BrandList from "../brands/BrandList";
import CategoryList from "../categories/CategoyList";
import TagLists from "../tags/TagList";
import Cart from "../cart/Cart";

beforeEach(commonBeforeEach)
afterEach(commonAfterEach)

describe('NavBar', () => {
  const store = configureStore({
    reducer: {
      cart: cartReducer
    }
  });

  test("renders without crashing", function () {
    render(
     <BrowserRouter>
      <Provider store={store}>
        <UserProvider>
          <NavBar />
        </UserProvider>
      </Provider>
    </BrowserRouter>
    );
  });
  
  test("matches snapshot", function () {
    const { asFragment } = render(
      <BrowserRouter>
        <Provider store={store}>
          <UserProvider>
            <NavBar />
          </UserProvider>
        </Provider>
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('Toggler component renders correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <Provider store={store}>
          <UserProvider>
            <Toggler />
          </UserProvider>
        </Provider>
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test('Cart component renders correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <Provider store={store}>
          <UserProvider>
            <Cart />
          </UserProvider>
        </Provider>
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test('BrandList component renders correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <UserProvider>
          <BrandList />
        </UserProvider>
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
  
  test('CategoryList component renders correctly', () => {
    const { container } = render(
      <BrowserRouter>
          <UserProvider>
            <CategoryList />
          </UserProvider>
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
    
  test('TagLists component renders correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <UserProvider>
          <TagLists />
        </UserProvider>
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});