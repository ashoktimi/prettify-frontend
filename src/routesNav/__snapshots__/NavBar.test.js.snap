// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar BrandList component renders correctly 1`] = `<div />`;

exports[`NavBar Cart component renders correctly 1`] = `<div />`;

exports[`NavBar CategoryList component renders correctly 1`] = `<div />`;

exports[`NavBar TagLists component renders correctly 1`] = `<div />`;

exports[`NavBar Toggler component renders correctly 1`] = `
<div>
  <nav
    class="navbar navbar-expand-bg navbar-light"
    style="padding: 0px;"
  >
    <div
      class="container-fluid"
    >
      <button
        aria-controls="offcanvasNavbar-expand"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="NavBar-name-div"
      >
        <a
          class="NavBar-name"
          href="/products"
        >
          Prettify
        </a>
      </div>
      <div
        aria-labelledby="offcanvasNavbarLabel-expand"
        class="offcanvas offcanvas-start"
        id="offcanvasNavbar-expand"
      >
        <h3
          class="Toggler-h3"
        >
          Please login to see.
        </h3>
      </div>
    </div>
  </nav>
</div>
`;

exports[`NavBar matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand navbar-light sticky-top"
    style="background-color: rgb(193, 111, 184); padding: 1rem; height: 4rem;"
  />
  <nav
    class="navbar navbar-expand-bg navbar-light"
    style="padding: 0px;"
  >
    <div
      class="container-fluid"
    >
      <button
        aria-controls="offcanvasNavbar-expand"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="NavBar-name-div"
      >
        <a
          class="NavBar-name"
          href="/products"
        >
          Prettify
        </a>
      </div>
      <div
        aria-labelledby="offcanvasNavbarLabel-expand"
        class="offcanvas offcanvas-start"
        id="offcanvasNavbar-expand"
      >
        <div
          class="offcanvas-header"
        >
          <div
            class="offcanvas-title h5"
            id="offcanvasNavbarLabel-expand"
            style="color: purple; left: 0px; right: 0px; margin: auto;"
          >
            Prettify
          </div>
          <button
            aria-label="Close"
            class="btn-close"
            type="button"
          />
        </div>
        <div
          class="offcanvas-body"
        >
          <ul
            class="Toggler-ul"
          >
            <li>
              <a
                href="/products"
              >
                ALL
              </a>
            </li>
            <li>
              <a
                href="/"
              >
                 BRANDS 
              </a>
            </li>
            <li>
              <a
                href="/"
              >
                CATEGORY
              </a>
            </li>
            <li>
              <a
                href="/"
              >
                TAGS
              </a>
            </li>
          </ul>
        </div>
        <a
          class="Toggler-bag"
          href="/"
        >
          <svg
            fill="currentColor"
            height="30"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M922.9 701.9H327.4l29.9-60.9 496.8-.9c16.8 0 31.2-12 34.2-28.6l68.8-385.1c1.8-10.1-.9-20.5-7.5-28.4a34.99 34.99 0 0 0-26.6-12.5l-632-2.1-5.4-25.4c-3.4-16.2-18-28-34.6-28H96.5a35.3 35.3 0 1 0 0 70.6h125.9L246 312.8l58.1 281.3-74.8 122.1a34.96 34.96 0 0 0-3 36.8c6 11.9 18.1 19.4 31.5 19.4h62.8a102.43 102.43 0 0 0-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7h161.1a102.43 102.43 0 0 0-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7H923c19.4 0 35.3-15.8 35.3-35.3a35.42 35.42 0 0 0-35.4-35.2zM305.7 253l575.8 1.9-56.4 315.8-452.3.8L305.7 253zm96.9 612.7c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 0 1-31.6 31.6zm325.1 0c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 0 1-31.6 31.6z"
            />
          </svg>
        </a>
        <a
          class="Toggler-logout"
          href="/"
          style="margin-bottom: 35px;"
        >
          Log out
        </a>
      </div>
    </div>
  </nav>
  <nav
    class="navbar navbar-expand-lg navbar-light sticky-top"
  >
    <div
      class="container"
    >
      <div
        class="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          class="mr-auto navbar-nav"
          style="left: 0px; right: 0px; margin: auto;"
        >
          <a
            class="NavBar-link"
            href="/products"
          >
            ALL
          </a>
          <a
            class="NavBar-link"
            href="/types/nail/nail_polish"
          >
            NAIL
          </a>
          <div
            class="NavBar-dropdown nav-item dropdown"
          >
            <a
              aria-expanded="false"
              class="dropdown-toggle nav-link"
              href="#"
              role="button"
              tabindex="0"
            >
              EYES
            </a>
          </div>
          <div
            class="NavBar-dropdown nav-item dropdown"
          >
            <a
              aria-expanded="false"
              class="dropdown-toggle nav-link"
              href="#"
              role="button"
              tabindex="0"
            >
              LIPS
            </a>
          </div>
          <div
            class="NavBar-dropdown nav-item dropdown"
          >
            <a
              aria-expanded="false"
              class="dropdown-toggle nav-link"
              href="#"
              role="button"
              tabindex="0"
            >
              FACE
            </a>
          </div>
          <div
            class="nav-item"
            style="padding-top: 5px;"
          >
            <a
              class="NavBar-link"
              href="/"
            >
              BRANDS
            </a>
          </div>
          <div
            class="nav-item"
            style="padding-top: 5px;"
          >
            <a
              class="NavBar-link"
              href="/"
            >
              CATEGORIES
            </a>
          </div>
          <div
            class="nav-item"
            style="padding-top: 5px;"
          >
            <a
              class="NavBar-link"
              href="/"
            >
              TAGS
            </a>
          </div>
          <a
            class="Navbar-bag"
            href="/"
          >
            <svg
              fill="currentColor"
              height="30"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M922.9 701.9H327.4l29.9-60.9 496.8-.9c16.8 0 31.2-12 34.2-28.6l68.8-385.1c1.8-10.1-.9-20.5-7.5-28.4a34.99 34.99 0 0 0-26.6-12.5l-632-2.1-5.4-25.4c-3.4-16.2-18-28-34.6-28H96.5a35.3 35.3 0 1 0 0 70.6h125.9L246 312.8l58.1 281.3-74.8 122.1a34.96 34.96 0 0 0-3 36.8c6 11.9 18.1 19.4 31.5 19.4h62.8a102.43 102.43 0 0 0-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7h161.1a102.43 102.43 0 0 0-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7H923c19.4 0 35.3-15.8 35.3-35.3a35.42 35.42 0 0 0-35.4-35.2zM305.7 253l575.8 1.9-56.4 315.8-452.3.8L305.7 253zm96.9 612.7c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 0 1-31.6 31.6zm325.1 0c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 0 1-31.6 31.6z"
              />
            </svg>
            <span
              class="bag-quantity"
            >
              0
            </span>
          </a>
          <div
            class="Profile-dropdown nav-item dropdown"
            style="margin-top: 10px;"
          >
            <a
              aria-expanded="false"
              class="dropdown-toggle nav-link"
              href="#"
              role="button"
              tabindex="0"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M16 9C16 11.2091 14.2091 13 12 13C9.79086 13 8 11.2091 8 9C8 6.79086 9.79086 5 12 5C14.2091 5 16 6.79086 16 9ZM14 9C14 10.1046 13.1046 11 12 11C10.8954 11 10 10.1046 10 9C10 7.89543 10.8954 7 12 7C13.1046 7 14 7.89543 14 9Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1ZM3 12C3 14.0902 3.71255 16.014 4.90798 17.5417C6.55245 15.3889 9.14627 14 12.0645 14C14.9448 14 17.5092 15.3531 19.1565 17.4583C20.313 15.9443 21 14.0524 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12ZM12 21C9.84977 21 7.87565 20.2459 6.32767 18.9878C7.59352 17.1812 9.69106 16 12.0645 16C14.4084 16 16.4833 17.1521 17.7538 18.9209C16.1939 20.2191 14.1881 21 12 21Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </a>
          </div>
          <a
            class="Navbar-logout"
            href="/"
          >
            Logout
          </a>
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
